returns/contrib/pytest/plugin.py

```diff
@@ -18,11 +21,9 @@
 )
 
 _FunctionType = TypeVar('_FunctionType', bound=Callable)
-# _ResultBasedType represents `Callable[[Any], ResultBasedN[Any, Any, Any]]`
-# but we cannot import `ResultBasedN` here because we're omitting this file from
-# the coverage report and `coverage.py` will exclude the class' coverage
-# information if we import it here.
-_ResultBasedType = TypeVar('_ResultBasedType', bound=Callable)
+_ResultCallableType = TypeVar(
+    '_ResultCallableType', bound=Callable[..., 'ResultBasedN'],
```

Comment:
Just a quick question: should we add `.trace` property to `ResultBasedN`?

Reply:
I think so, but our feature `collect_traces` will not work with custom Results because we monkey patch `_Failure` directly!

Reply:
Ok, then it should not be a part of the interface 👍

Reply:
I'll make some tests latter, I think we can achieve the trace for custom Results by monkey patching `ResultBasedN` directly!!
But the trace will be available in all types like `_Success`, it can be a problem or not since `collect_traces` should be used only for development purposes.

Reply:
I don't think that we need this. Because this is an implementation detail. I am not sure that other people might need it.